### PR TITLE
added InputGroup

### LINKED ISSUE #251: Create new Input Group Component
Screen Shot 2022-02-16 at 16 38 12

### PR DESCRIPTION
**Pull request by linuscrypto**
Closes #251

### CODE REVIEW

File: components/Input/InputGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		>
+			<StyledInput {...rest} />
+			<StyledIcon ref={iconRef as any}>{icon}</StyledIcon>
+		</InputWrapper>
+	);
+};
+
+const StyledInput = styled(Input)``;
+
+const StyledIcon = styled.div``;
```
Comment (saeta-eth) [inline components/Input/InputGroup.tsx:33]: Necessary? You can use `div` and `Input` directly if they don't use any styling.

File: components/Input/InputGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+export const InputGroup: FC<IInputGroup> = ({ icon, iconPosition = 'right', ...rest }) => {
+	const iconRef = useRef<HTMLInputElement>();
+
+	return (
+		<InputWrapper
+			className="inputGroup"
+			iconPosition={iconPosition}
+			iconSize={iconRef?.current?.getBoundingClientRect()?.width}
```
Comment (saeta-eth) [inline components/Input/InputGroup.tsx:23]: Nice

File: components/Input/InputGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		font-size: 14px;
+		font-family: ${(props) => props.theme.fonts.agrandir};
+	}
+
+	& > ${StyledInput} {
+		padding-right: ${(props) => (props.iconPosition === 'right' ? `${props.iconSize}px` : '8px')};
+		padding-left: ${(props) => (props.iconPosition === 'left' ? `${props.iconSize}px` : '8px')};
+	}
+
+	&.inputGroup {
```
Comment (saeta-eth) [inline components/Input/InputGroup.tsx:64]: I think you already made it `position:relative` in the [first line](https://github.com/AelinXYZ/aelin-ui/pull/253/files#diff-3c49a0c8128324d5c74a5aef4b70e4e474d2f6eed8429ebc7585e9bbac14cab8R36). Right? if so, you can also remove the [className="inputGroup"](https://github.com/AelinXYZ/aelin-ui/pull/253/files#diff-3c49a0c8128324d5c74a5aef4b70e4e474d2f6eed8429ebc7585e9bbac14cab8R21)

Comment (linuscrypto) [reply components/Input/InputGroup.tsx:64]: good catch!

File: components/Input/InputGroup.tsx (hunk 1 above)
Comment (alextheboredape) [reply components/Input/InputGroup.tsx:32]: I am not sure you need it to be wrapped. Does it not work if you just use the Input or a div tag in the bottom section? I dont think this is a huge issue though imo. It is fairly minor to have the extra lines if it is a bit easier to make it work

Comment (alextheboredape) [reply components/Input/InputGroup.tsx:32]: just using a div might be too ambiguous if the icon is a div as I can see below. I think its prob ok to leave it wrapped

**Review by alextheboredape — APPROVED** [review]

File: components/Input/InputGroup.tsx (hunk 1 above)
Comment (linuscrypto) [reply components/Input/InputGroup.tsx:32]: Just tried out and worked without being wrapped... :)

**Review by saeta-eth — APPROVED** [review]

**Pull request merged**